dev/README_AIRFLOW3_DEV.md

```diff
@@ -0,0 +1,132 @@
+<!--
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+-->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of contents**
+
+- [Main branch is Airflow 3](#main-branch-is-airflow-3)
+- [Contributors](#contributors)
+  - [Developing for providers and Helm chart](#developing-for-providers-and-helm-chart)
+  - [Developing for Airflow 3 and 2.10.x / 2.11.x](#developing-for-airflow-3-and-210x--211x)
+  - [Developing for Airflow 3](#developing-for-airflow-3)
+  - [Developing for Airflow 2.10.x](#developing-for-airflow-210x)
+  - [Developing for Airflow 2.11](#developing-for-airflow-211)
+- [Committers / PMCs](#committers--pmcs)
+  - [Merging PRs for providers and Helm chart](#merging-prs-for-providers-and-helm-chart)
+  - [Merging PR for Airflow 3 and 2.10.x / 2.11.x](#merging-pr-for-airflow-3-and-210x--211x)
+  - [Merging PRs 2.10.x](#merging-prs-210x)
+  - [Merging PRs for Airflow 3](#merging-prs-for-airflow-3)
+  - [Merging PRs for Airflow 2.11](#merging-prs-for-airflow-211)
+- [Milestones for PR](#milestones-for-pr)
+  - [Set 2.10.x milestone](#set-210x-milestone)
+  - [Set 2.11 milestone](#set-211-milestone)
+  - [Set 3 milestone](#set-3-milestone)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Main branch is Airflow 3
+
+The main branch is for development of Airflow 3.
+Airflow 2.10.x releases will be cut from `v-2-10-stable` branch.
+
+# Contributors
+
+The following section explains which branches you should target your PR to
+
+## Developing for providers and Helm chart
+
+PRs should target `main` branch.
+Make sure your code is only about Providers or Helm chart.
+Avoid mixing core changes into the same PR
+
+## Developing for Airflow 3 and 2.10.x / 2.11.x
+
+If the PR is relevant for both Airflow 3 and 2 it should target `main` branch.
```

Comment:
Do we/shall we define a label to mark PRs which are going into both? Plus shall we add a note for contributors coming along?

```suggestion
If the PR is relevant for both Airflow 3 and 2 it should target `main` branch.

Note that Airflow 2 is not planned having a feature release other than a bridge release. If you plan to add features you should align with contributors first.
```

Reply:
Please check the updated doc
I improved this part with the mental model